### PR TITLE
Merge meta data in setup.cfg into pyproject.toml

### PR DESCRIPTION
**Pull request by KyleQianliMa**
# Short description of the changes:

Merging meta data in setup.cfg into pyproject.toml.
This is to address [Story 3163](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=3163)
# Long description of the changes:

# Check list for the pull request
- [ ] I have read the [CONTRIBUTING]
- [ ] I have read the [CODE_OF_CONDUCT]
- [ ] I have added tests for my changes
- [ ] I have updated the documentation accordingly

# Check list for the reviewer
- [ ] I have read the [CONTRIBUTING]
- [ ] I have verified the proposed changes
- [ ] best software practices
    + [ ] all internal functions have an underbar, as is python standard
    + [ ] clearly named variables (better to be verbose in variable names)
    + [ ] code comments explaining the intent of code blocks
- [ ] All the tests are passing
- [ ] The documentation is up to date
- [ ] code comments added when explaining intent

# Manual test for the reviewer

# References

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,22 +1,48 @@
 [project]
-name = "LiquidsReflectometer"
+name = "lr_reduction"
+authors = [
+	{name = "Mat", email = "m2d@ornl.gov"}
+]
 description = "Reduction scripts for the Liquids Reflectometer. This includes both automated reduction scripts and useful scripts to reprocess data."
-dynamic = ["version", "readme", "license", "dependencies"]
+dynamic = ["version"]
```
Comment (jmborr) [inline pyproject.toml:7]:
```suggestion
version = "0.1.0"  # initial version, dynamically overriden
```

File: pyproject.toml [hunk 2]
```diff
@@ -1,22 +1,48 @@
 [project]
-name = "LiquidsReflectometer"
+name = "lr_reduction"
+authors = [
+	{name = "Mat", email = "m2d@ornl.gov"}
```
Comment (jmborr) [inline pyproject.toml:4]: change tab to spaces

File: pyproject.toml [hunk 3]
```diff
@@ -1,22 +1,48 @@
...
 classifiers = [
   "Operating System :: OS Independent",
   "Programming Language :: Python :: 3"
 ]
+dependencies = [
+    "lmfit",
+    "numpy",
+    "matplotlib",
+    "mantid"
+
```
Comment (jmborr) [inline pyproject.toml:20]: remove the empty line

File: pyproject.toml [hunk 4]
```diff
@@ -1,22 +1,48 @@
...
 description = "Reduction scripts for the Liquids Reflectometer. This includes both automated reduction scripts and useful scripts to reprocess data."
-dynamic = ["version", "readme", "license", "dependencies"]
+dynamic = ["version"]
+license = {file = "LICENSE"}
+readme = "README.md"
 requires-python = ">=3.8"
 classifiers = [
   "Operating System :: OS Independent",
   "Programming Language :: Python :: 3"
 ]
```
Comment (jmborr) [inline pyproject.toml:14]: "Topic :: Scientific/Engineering :: Neutron reflectometry"

File: pyproject.toml [hunk 5]
```diff
@@ -1,22 +1,48 @@
...
+    "mantid"
+
+]
+
+[project.optional-dependencies]
+tests = [
+    "mock",
+    "pytest",
+    "pytest-qt",
+    "typing"
```
Comment (jmborr) [inline pyproject.toml:28]: remove `typing`, `mock`, and `pytest-qt`

**Review by jmborr — CHANGES_REQUESTED** [review]
did you try installing the package with `pip install -e .`?

**Comment by KyleQianliMa** [comment]
added 
[tool.setuptools.packages.find]
where = ["reduction"]
exclude = ["tests*", "scripts*", "docs*", "notebooks*"] in pyproject.toml and addressed other comments.

**Review by jmborr — APPROVED** [review]

**Comment by KyleQianliMa** [comment]
I think the conflict has been resolved and now ready to merge

**Review by jmborr — APPROVED** [review]
`pip install -e .` now installs package `lr_reduction` so that later one can do `import lr_reduction`.

**Pull request merged**